tests/php/Model/EditableFormFieldTest.php

```diff
@@ -168,6 +168,28 @@ public function testFileFieldAllowedExtensionsBlacklist()
         $this->assertNotContains('jpg', $formField->getValidator()->getAllowedExtensions());
     }
 
+    /**
+     * Test that if folder is not set or folder was removed,
+     * then getFormField() saves file in protected folder
+     */
+    public function testFileFieldSaveInFolder()
+    {
+        $fileField_1 = $this->objFromFixture(EditableFileField::class, 'file-field-without-folder');
+        $fileField_2 = $this->objFromFixture(EditableFileField::class, 'file-field-with-folder');
+
+        $formField_1 = $fileField_1->getUploadFormField();
+        $formField_2 = $fileField_2->getFormField();
+
+        $canViewParent_1 = $fileField_1->Folder()->Parent()->CanViewType;
+        $canViewParent_2 = $fileField_2->Folder()->Parent()->CanViewType;
+
+        $this->assertEquals('OnlyTheseUsers', $canViewParent_1);
+        $this->assertEquals('Inherit', $canViewParent_2);
+
+        $this->assertTrue((bool) preg_match('/^Form-submissions\/0_+/', $formField_1->folderName));
```

Comment:
Create a UserDefinedForm page, either in fixtures or just quickly create and write it in this unit test.  Update the ParentID on $formField_1 to this.  Update the regex to

`sprintf('/^Form-submissions\/%d_%d/', $udf->ID, $formField_1->ID)`